### PR TITLE
docs: Update iOS Linking information

### PR DESCRIPTION
**Pull request by clucasalcantara**
### Summary:

After the discussions on issue #56 I've noticed that we have missing information about iOS Linking and almost none about the manual linking process, this PR aims to provide this missing info.

### Test Plan:

No needed, only docs changes.

### CODE REVIEW

File: docs/Linking.md [hunk 1]
```diff
@@ -1,6 +1,19 @@
 # Manual Linking
 
 ## iOS
+1. Open your project `.xcodeproj`
```
Comment (Krizzu) [inline docs/Linking.md]: Please split `.xcodeproj` linking and `pods`, as they're two separate methods to link it.

Comment (clucasalcantara) [reply docs/Linking.md]: Hi @Krizzu! :) Thanks for taking a look at this :)

I've changed the language a little bit and I think now the steps are more clear to follow.

**Comment by clucasalcantara** [comment]
Hey @Krizzu :) Noted! I've made a few changes, please take a look :)

**Comment by Krizzu** [comment]
Hey @clucasalcantara,

Can you split `pod` and `.xcodeproj` linking into sections please?


thanks.

**Comment by clucasalcantara** [comment]
> Hey @clucasalcantara,
> 
> Can you split `pod` and `.xcodeproj` linking into sections please?
> 
> thanks.

Sorry @Krizzu , didn't noticed that :) I've split into step one and step two.

**Comment by Krizzu** [comment]
Hey @clucasalcantara 

I would not call them `step one/two` as this is misleading. Those are completely separate steps, not requiring one another.

You choose one of the way to link a library, otherwise you end up with same module, linked twice.


thanks

**Comment by clucasalcantara** [comment]
Hey @Krizzu 

`XCode Mode` and `Pod Mode`  sounds better?

**Comment by Krizzu** [comment]
Hi @clucasalcantara 

What about `Using 'Pods'` and `Project linking` ?

**Comment by clucasalcantara** [comment]
Even better, on it

File: docs/Linking.md [hunk 2]
```diff
@@ -2,6 +2,27 @@
...
+
+3. Go to `Build Phases -> Link Binary with Libraries` and add:  `libRNCAsyncStorage.a`.
+
+#### Using 'Pods'
+1. Enter into iOS Folder `cd ios/` (on your project's root folder).
+
+2. Add this line to your `Podfile` just below the last pod: (if you don't one just create a new runnning: `pod init`).
+
+```
+pod 'react-native-async-storage', :path => '../node_modules/@react-native-community/async-storage'
```
Comment (Krizzu) [inline docs/Linking.md]: Let's including diffing, like in Android section:
```
	```diff
	+ pod 'react-native-async-storage', :path => '../node_modules/@react-native-community/async-storage'
	```
```

File: docs/Linking.md [hunk 3]
```diff
@@ -2,6 +2,27 @@
 
 ## iOS
 
+#### Project linking
+1. Open your project `.xcodeproj` on xcode.
+
+2. Right click on the Libraries folder and select `Add files to "yourProjectName"`.
+
+3. Add `RNCAsyncStorage.xcodeproj` to your project Libraries.
```
Comment (Krizzu) [inline docs/Linking.md]: Mind adding where we can find that `.xcodeproj` file? 
Location: `node_modules/@react-native-community/async-storage/ios`

File: docs/Linking.md (hunk 2 above)
Comment (clucasalcantara) [reply docs/Linking.md]: Done :)

File: docs/Linking.md (hunk 3 above)
Comment (clucasalcantara) [reply docs/Linking.md]: Also done 💪

File: README.md [hunk 4]
```diff
@@ -12,6 +12,9 @@ $ yarn add @react-native-community/async-storage
 
 # Link
 $ react-native link @react-native-community/async-storage
+
+# Pod installation
+$ cd ios/ && pod install
```
Comment (Krizzu) [inline README.md]: Can you move this bit below (just above "See docs...") and note it's only for `iOS` project with `pods`?

Something like:
```markdown
**Note:** For iOS project using `pods`, run:
	```
	$ cd ios/ && pod install
	```
```

Comment (clucasalcantara) [reply README.md]: Done, also!

**Pull request merged**